api/agent/pure_runner.go

```diff
@@ -559,13 +584,35 @@ func (pr *pureRunner) AddCallListener(cl fnext.CallListener) {
 	pr.a.AddCallListener(cl)
 }
 
+func (pr *pureRunner) saveCallHandle(ch *callHandle) {
+	pr.callHandleLock.Lock()
+	pr.callHandleMap[ch.c.Model().ID] = ch
+	pr.callHandleLock.Unlock()
+}
+
+func (pr *pureRunner) removeCallHandle(cID string) {
+	pr.callHandleLock.Lock()
+	delete(pr.callHandleMap, cID)
+	pr.callHandleLock.Unlock()
+
+}
+
 func (pr *pureRunner) spawnSubmit(state *callHandle) {
 	go func() {
 		err := pr.a.Submit(state.c)
 		state.enqueueCallResponse(err)
 	}()
 }
 
+func (pr *pureRunner) spawnDetachSubmit(state *callHandle) {
+	go func() {
+		pr.saveCallHandle(state)
+		err := pr.a.Submit(state.c)
+		pr.removeCallHandle(state.c.Model().ID)
+		state.enqueueCallResponse(err)
```

Comment:
i don't think we need, the lb_agent will retrun to the client with a 202 or a 50x error, then the actual execution of the function is executed on a separate routine and once it will complete we shut down everything as it was a regular call. I can draw a graph of what I think this works so you can see if it makes sense to you and if I miss something important